### ISSUE TITLE
[BUG] toBeVisible With 100 minute timeout Fails After 2ms

### ISSUE THREAD

**Issue by jwalkerinterpres** [microsoft/playwright#28936]
### System info
- Playwright Version: ^1.40.1
- Operating System: MacOS 14.2.1 
- Browser: Chromium (and All?)
- Other info:

### Source code

The key part is:

    const notification = page.locator(
        `div[data-projection-id] div[data-projection-id] div ` +
        `div:nth-child(1) div:has-text("Integration Created")`
    );
    await expect(notification, 'integration save notification').toBeVisible({
      timeout: 100 * 60 * 1000
    });

When I run a test with that code, it fails on the expect ... after taking only 2ms.

I'm running Playwright with the `--ui` flag, and in the Errors tab it says:

```
expect(locator).toBeVisible()

Locator: locator('div[data-projection-id] div[data-projection-id] div div:nth-child(1) div:has-text("Integration Created")')
Expected: visible
Received: hidden
Call log:
  - integration save notification with timeout 6000000ms
  - waiting for locator('div[data-projection-id] div[data-projection-id] div div:nth-child(1) div:has-text("Integration Created")')
```

"with timeout 6000000ms" suggests that the timeout was received ... and yet the Call tab shows:

> Error: integration save notification [2mexpect([22m[31mlocator[39m[2m).[22mtoBeVisible[2m()[22m Locator: locator('div[data-projection-id] div[data-projection-id] div div:nth-child(1) div:has-text("Integration Created")') Expected: visible Received: hidden Call log: [2m- integration save notification with timeout 6000000ms[22m [2m - waiting for locator('div[data-projection-id] div[data-projection-id] div div:nth-child(1) div:has-text("Integration Created")')[22m
> TIME
> wall time:
> 1/9/2024, 4:49:51 PM
> duration:
> 2ms

A duration of 2ms !== 6000000ms

**Steps**
- [Run the test]
- That step should take 6000000ms before failing

**Expected**

When I pass a timeout to `toBeVisible`, it should wait that time ... not 2 ms. 

**Actual**

It waits 2ms.

**Comment by jwalkerinterpres** [comment]
So, I wrote this ticket last night, then finished it up and submitted it.  Then I re-ran my test (with _nothing_ changed) ... and it waited ten minutes for that step!

To be clear, this is not "the user is an idiot and forgot to save"; I definitely saved the file last night.  However, some old code (without the ten minute wait) was evidently preventing my new timeout from working (even though it _was_ received, as you can see from my Errors tab output).

In short, it seems there's some sort of temporary caching (or something similar) bug in Playwright ... but I can no longer reproduce it, so I don't know how useful this bug report will be :(

**Comment by jwalkerinterpres** [comment]
I just tried again and thought it was working ... but it turned out it was just Playwright hanging :(. I tried again and it failed too quickly again.

Interestingly, on this latest run the Call tab claims the step only had a "duration: 0ms" (not even 2ms!)

**Comment by jwalkerinterpres** [comment]
Here are screenshots of the two tabs.

<img width="751" alt="image" src="https://github.com/microsoft/playwright/assets/152227747/18a52532-1959-422c-a354-db9334d355ed">
<img width="1217" alt="image" src="https://github.com/microsoft/playwright/assets/152227747/0ec07030-eb7a-42bf-bfa7-d09770a1b744">

**Comment by jwalkerinterpres** [comment]
One more quick note: I fixed the selector and the test works fine now, so evidently this only happens when the selector is for a non-existent (ie. not visible) element.

**Comment by dgozman** [comment]
Could you please provide a self-contained repro that we can run locally? Otherwise we won't be able to debug and fix the problem.

**Comment by jwalkerinterpres** [comment]
I can't unfortunately: this is happening in a closed-source app, and it would be expensive (time-wise) to create a static HTML page of it.  I provided the relevant code that's causing it, as well as screenshots though.

If nothing else, now you're aware that this is a thing that can happen ... and if you have to close the ticket for not being reproducible, that's reasonable.

**Comment by dgozman** [comment]
@jwalkerinterpres Since this is a first time we see this issue, a repro would be essential. All our timeout-related tests pass, so the problem is either in your tests/setup, or there is an edge case that we are not aware of. Both require a repro so that we can debug the problem.

Let me close this issue for now, but don't hesitate to file a new one if you come up with a repro.
